### PR TITLE
chore: use TestContainer for unit tests

### PR DESCRIPTION
**Pull request by phiz71**
Closes gravitee-io/issues#3926

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -35,7 +35,7 @@
 		<gravitee-repository.version>3.9.0-SNAPSHOT</gravitee-repository.version>
 		<gravitee-repository-test.version>3.9.0-SNAPSHOT</gravitee-repository-test.version>
 		<gravitee-definition.version>1.25.0</gravitee-definition.version>
-		<embed.mongo.version>2.0.0</embed.mongo.version>
+        <testcontainers.version>1.15.3</testcontainers.version>
```
Comment (ytvnr) [inline pom.xml:38]: formatting ?

File: src/test/java/io/gravitee/repository/mongodb/MongoTestRepositoryConfiguration.java [hunk 2]
```diff
@@ -32,27 +36,33 @@
 @EnableMongoRepositories
 public class MongoTestRepositoryConfiguration extends AbstractRepositoryConfiguration {
 
-    @Bean
-    public MongoTestFactory factory() throws Exception {
-        return MongoTestFactory.with(Version.Main.V3_5);
+    @Inject
+    private MongoDBContainer embeddedMongoDb;
```
Comment (ytvnr) [inline src/test/java/io/gravitee/repository/mongodb/MongoTestRepositoryConfiguration.java:40]:
```suggestion
    private MongoDBContainer mongoDbContainer;
```

File: src/test/java/io/gravitee/repository/mongodb/MongoTestRepositoryConfiguration.java [hunk 3]
```diff
@@ -32,27 +36,33 @@
...
+
+    @Bean(destroyMethod = "stop")
+    public MongoDBContainer embeddedMongoDb() {
```
Comment (ytvnr) [inline src/test/java/io/gravitee/repository/mongodb/MongoTestRepositoryConfiguration.java:43]:
```suggestion
    public MongoDBContainer mongoDbContainer() {
```

**Review by ytvnr — COMMENTED** [review]
little comments

**Review by ytvnr — APPROVED** [review]

File: src/test/java/io/gravitee/repository/mongodb/MongoTestRepositoryInitializer.java [hunk 4]
```diff
@@ -24,20 +25,33 @@
...
 
     private static final Logger LOG = LoggerFactory.getLogger(MongoTestRepositoryInitializer.class);
 
+    private final MongoClient mongoClient;
+
     @Autowired
-    private MongoTemplate mongoTemplate;
+    public MongoTestRepositoryInitializer(MongoClient mongoClient) {
+        LOG.debug("Constructed");
+        LOG.debug("Dropping database...");
```
Comment (gaetanmaisse) [inline src/test/java/io/gravitee/repository/mongodb/MongoTestRepositoryInitializer.java:40]: Hmm, sure?

File: src/test/java/io/gravitee/repository/mongodb/MongoTestRepositoryConfiguration.java [hunk 5]
```diff
@@ -32,27 +36,33 @@
...
 
-    @Bean
-    public MongoTestFactory factory() throws Exception {
-        return MongoTestFactory.with(Version.Main.V3_5);
+    @Inject
+    private MongoDBContainer mongoDBContainer;
+
+    @Bean(destroyMethod = "stop")
+    public MongoDBContainer mongoDBContainer() {
+        MongoDBContainer mongoDb = new MongoDBContainer(DockerImageName.parse("mongo:4.0.10"));
```
Comment (gaetanmaisse) [inline src/test/java/io/gravitee/repository/mongodb/MongoTestRepositoryConfiguration.java:44]: Why `4.0.10` specifically?

**Review by gaetanmaisse — APPROVED** [review]

**Pull request merged**